OpenNMT/train.py

```diff
@@ -254,42 +287,47 @@ def main():
 
     print('Building model...')
 
-    if opt.train_from is None:
-        encoder = onmt.Models.Encoder(opt, dicts['src'])
-        decoder = onmt.Models.Decoder(opt, dicts['tgt'])
-        generator = nn.Sequential(
-            nn.Linear(opt.rnn_size, dicts['tgt'].size()),
-            nn.LogSoftmax())
-        if opt.cuda > 1:
-            generator = nn.DataParallel(generator, device_ids=opt.gpus)
-        model = onmt.Models.NMTModel(encoder, decoder, generator)
-        if opt.cuda > 1:
-            model = nn.DataParallel(model, device_ids=opt.gpus)
-        if opt.cuda:
-            model.cuda()
-        else:
-            model.cpu()
-
-        model.generator = generator
+    encoder = onmt.Models.Encoder(opt, dicts['src'])
+    decoder = onmt.Models.Decoder(opt, dicts['tgt'])
 
+    generator = nn.Sequential(
+        nn.Linear(opt.rnn_size, dicts['tgt'].size()),
+        nn.LogSoftmax())
+
+    model = onmt.Models.NMTModel(encoder, decoder)
+
+    if opt.train_from:
```

Comment:
I guess I'm overruled about using `load_state_dict`. Notice that now you might accidently load the state from a model with different options (e.g. unidirectional vs. bidirectional), with undefined results (maybe some random error, maybe silently giving bad results).

Also, this means that everyone's checkpoints will be broken. You'll at least need to go fix the checkpoints mentioned in the README, sorry.

Also please use `if opt.train_from is not None`, it's safer in Python.